examples/kyb-app/src/components/organisms/DynamicUI/StateManager/hooks/useStateLogic/useStateLogic.ts

```diff
@@ -0,0 +1,111 @@
+import { AnyObject } from '@ballerine/ui';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import isEqual from 'lodash/isEqual';
+import { StateMachineAPI } from '@app/components/organisms/DynamicUI/StateManager/hooks/useMachineLogic';
+import { getAccessToken } from '@app/helpers/get-access-token.helper';
+import { useDynamicUIContext } from '@app/components/organisms/DynamicUI/hooks/useDynamicUIContext';
+import { CollectionFlowContext } from '@app/domains/collection-flow/types/flow-context.types';
+
+interface State {
+  machineState: string;
+  payload: CollectionFlowContext;
+}
+
+export const useStateLogic = (machineApi: StateMachineAPI, initialContext = {}) => {
+  const [contextPayload, setState] = useState<State>(() => ({
+    machineState: machineApi.getState(),
+    payload: machineApi.getContext() as CollectionFlowContext,
+  }));
+
+  const contextRef = useRef<State>(contextPayload);
+  const { helpers } = useDynamicUIContext();
+  const host = new URL(import.meta.env.VITE_API_URL as string).host;
+  const protocol = new URL(import.meta.env.VITE_API_URL as string).protocol;
```

Comment:
Environment variables should have a validation schema. Asserting to `string` here does not cover the cases where the env variable is invalid.